layerindex/tools/github-fetch.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+# Github fetch utility
+
+# Copyright (C) 2017 Intel Corporation
+#
+# Licensed under the MIT license, see COPYING.MIT for details
+#
+# SPDX-License-Identifier: MIT
+
+import sys
+import os
+import argparse
+import json
+import requests
+import re
+import subprocess
+
+def fetchall(args):
+    link_re = re.compile('<(http[^>]+)>; rel="([a-zA-Z0-9]+)"')
+
+    keys = {'orgname': args.organisation,
+            'access_token': args.access_token,
+            'per_page': 100
+            }
+    url = 'https://api.github.com/orgs/{orgname}/repos?access_token={access_token}&per_page={per_page}'
+    url = url.format(**keys)
+
+    existing = os.listdir(args.outdir)
+    for name in existing:
+        if name.endswith('.deleted'):
+            print('Directories marked deleted (suffix .deleted) still exist in output path - remove these to continue')
+            sys.exit(1)
+
+    if args.resume_from:
+        fetching = False
+    else:
+        fetching = True
+
+    failed = []
+    repos = None
+    while True:
+        session = requests.Session()
+        print('getting %s' % url)
+        with requests.Session() as s:
+            r = s.get(url)
+            if not r.ok:
+                print('Request failed: %d: %s' % (r.status_code, r.text))
+                sys.exit(2)
+            st = r.text
+            repos = json.loads(st)
+            for repo in repos:
+                name = repo['name']
+                if name in existing:
+                    existing.remove(name)
+                if args.resume_from and name == args.resume_from:
+                    fetching = True
+                elif not fetching:
+                    print('Skipping %s' % name)
+                    continue
+                clone_url = repo['clone_url']
+                outpath = os.path.join(args.outdir, name)
+                if os.path.exists(outpath):
+                    print('Update %s' % outpath)
+                    ret = subprocess.call(['git', 'fetch'], cwd=outpath)
+                    if ret == 0:
+                        ret = subprocess.call(['git', 'reset', '--hard', 'FETCH_HEAD'], cwd=outpath)
+                else:
+                    print('Fetch %s' % clone_url)
+                    ret = subprocess.call(['git', 'clone', clone_url, name], cwd=args.outdir)
+                if ret != 0:
+                    failed.append(name)
+
+            link = r.headers.get('Link', None)
+            if link:
+                linkitems = dict([reversed(x) for x in link_re.findall(link)])
+                url = linkitems.get('next', None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)